src/dataflow/src/render/reduce.rs

```diff
@@ -1088,10 +1095,12 @@ where
                 // Should negative accumulations reach us, we should loudly complain.
                 if source.iter().any(|(_val, cnt)| cnt <= &0) {
                     for (val, cnt) in source.iter() {
-                        if cnt <= &0 {
-                            // XXX: This reports user data, which we perhaps should not do!
-                            log::error!("[customer-data] Non-positive accumulation in MinsMaxesHierarchical: key: {:?}\tvalue: {:?}\tcount: {:?}", key, val, cnt);
-                        }
+                        // XXX: This reports user data, which we perhaps should not do!
+                        assert_or_log!(
+                            *cnt > 0,
```

Comment:
this is not part of Nikhil's change but I forget why the assertion here is "has to be positive" whereas elsewhere its "has to be zero or positive". should this be changed to >= 0 instead?